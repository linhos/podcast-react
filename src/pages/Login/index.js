import React, {useRef, useState} from 'react';
import {useHistory} from 'react-router-dom';
import loginUser from '../../services/loginUser';


const Login = () => {
    const history = useHistory();
    const loginForm = useRef(null)
    const [email, setEmail] = useState('');
    const [password, setPassword] = useState(false)

    const handleSubmit = (event) => {
        event.preventDefault();
        setEmail(loginForm.current.email.value);
        setPassword(loginForm.current.password.value);
        const result = loginUser(email, password);
        if (result.user) {
            history.push(`/dashboard`);
        }
    }


    return(
        <>
            <h1>Login</h1>
            <form ref={loginForm} onSubmit={handleSubmit}>
                <label htmlFor='email'>Usuario</label>
                <input type="text" name="email" id="email" required/>
                <hr />
                <label htmlFor='password'>Contraseña</label>
                <input type="password" name="password" id="password" required />
                <br /><br />
                <input type="submit" value="Inicia sesión" />
            </form>
        </>
    )
}

export default Login;