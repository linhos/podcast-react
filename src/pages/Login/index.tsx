import React, { useRef, useState } from "react";
import { useHistory } from "react-router-dom";
import loginUser from "../../services/loginUser";

const Login: React.FC = () => {
  const history = useHistory();
  const loginForm = useRef(null);
  const [email, setEmail] = useState('');
  const [emailError, setEmailError] = useState('');
  const [password, setPassword] = useState('');
  const [passwordError, setPasswordError] = useState('');
  const [validationError, setValidationError] = useState('');

  const handleLoginSubmit = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    if (!email) {
      setEmailError("El usuario es obligatorio.");
    }
    if (!password) {
      setPasswordError("La contraseña es obligatoria.");
    }

    if (email && password) {
      const result = loginUser(email, password);

      if(result.errors.length !== 0) {
        setValidationError('Los datos proporcionados no son válidos.')
      } else {
        history.push(`/dashboard`);
      }
    }
  };

  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value:string = e.target.value;
    setEmail(value);
  };

  const handlePasswordChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const value:string = e.target.value;
    setPassword(value);
  };

  return (
    <>
      <h1>Login</h1>
      <form ref={loginForm} onSubmit={handleLoginSubmit}>
        <label htmlFor="email">Usuario</label>
        <input
          type="text"
          name="email"
          id="email"
          onChange={handleEmailChange}
          placeholder="usuario"
        />
        {emailError && (
          <div data-testid="email-errors" className="error">
            {emailError}
          </div>
        )}
        <hr />
        <label htmlFor="password">Contraseña</label>
        <input
          type="password"
          name="password"
          id="password"
          onChange={handlePasswordChange}
          placeholder="contraseña"
        />
        {passwordError && (
          <div data-testid="password-errors" className="error">
            {passwordError}
          </div>
        )}
        {validationError && (
          <div data-testid="validation-errors" className="error">
            {validationError}
          </div>
        )}
        <br />
        <br />
        <input type="submit" value="Inicia sesión" />
      </form>
    </>
  );
};

export default Login;
