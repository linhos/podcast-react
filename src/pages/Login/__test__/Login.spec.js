import React from 'react';
import {render, cleanup} from '@testing-library/react';


import Login from '../index';

describe('test suite for Login Component', () => {
    afterEach(cleanup);
    it('render the form correctly', () => {
        
        const {getByLabelText, getByText} = render(<Login />);

        const usernameInput = getByLabelText('Usuario');
        expect(usernameInput).toHaveAttribute('type', 'text');
        expect(usernameInput).toHaveAttribute('required');
        expect(usernameInput).not.toHaveValue();

        const passwordInput = getByLabelText('Contraseña')
        expect(passwordInput).toHaveAttribute('type', 'password');
        expect(passwordInput).toHaveAttribute('required');

        const buttonSubmit = getByText('Inicia sesión');
        expect(buttonSubmit).toHaveValue('Inicia sesión');
        
    })

    it('renders', () => {
        const {asFragment} = render(<Login />);
        console.log(asFragment());

        expect(asFragment()).toMatchSnapshot();
    })
})