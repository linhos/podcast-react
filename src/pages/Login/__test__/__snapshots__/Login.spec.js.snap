// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for Login Component renders 1`] = `
<DocumentFragment>
  <h1>
    Login
  </h1>
  <form>
    <label
      for="email"
    >
      Usuario
    </label>
    <input
      id="email"
      name="email"
      required=""
      type="text"
    />
    <hr />
    <label
      for="password"
    >
      Contraseña
    </label>
    <input
      id="password"
      name="password"
      required=""
      type="password"
    />
    <br />
    <br />
    <input
      type="submit"
      value="Inicia sesión"
    />
  </form>
</DocumentFragment>
`;
